sdk/core/azure-core/inc/azure/core/http/policy.hpp

```diff
@@ -390,22 +391,46 @@ namespace Azure { namespace Core { namespace Http {
         NextHttpPolicy policy) const override;
   };
 
+  /**
+   * @brief Options for Azure::Core::Http::LogPolicy.
+   */
+  struct LogOptions
+  {
+    /**
+     * @brief HTTP query parameters that are allowed to be logged.
+     * @remark URL-encoded parameter names are expected.
+     */
+    std::set<std::string> AllowedHttpQueryParameters;
```

Comment:
Do we have an example code where some client library or unit tests would set these and what the outcome would be?

Presumably we want to follow the similar defaults for storage and other client libraries like .NET, once storage starts adding logging.

How would the end user customize which headers to log?

Reply:
Should we turn these headers on by default?

https://github.com/Azure/azure-sdk-for-net/blob/d9a83c970d456c4d93c308869d44a84b531f44a2/sdk/core/Azure.Core/src/DiagnosticsOptions.cs#L24-L49

@tg-msft what do you think?

Reply:
Yes, and Storage should add https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Blobs/src/BlobClientOptions.cs#L128 which we worked carefully with their team on (and used across all the other languages).

Reply:
@tg-msft do you expect users to pass in query parameter strings for the allow list that already URL-encoded? The docs don't specify.
https://github.com/Azure/azure-sdk-for-net/blob/d9a83c970d456c4d93c308869d44a84b531f44a2/sdk/core/Azure.Core/src/DiagnosticsOptions.cs#L85-L88

And why do you ignore casing when comparing headers and query parameter names?

https://github.com/Azure/azure-sdk-for-net/blob/675631c134f1ae01fe170bb3815f29cec96ea1fc/sdk/core/Azure.Core/src/Shared/HttpMessageSanitizer.cs#L27
https://github.com/Azure/azure-sdk-for-net/blob/675631c134f1ae01fe170bb3815f29cec96ea1fc/sdk/core/Azure.Core/src/Shared/HttpMessageSanitizer.cs#L100

Reply:
I don't know if Azure has any query parameter names that need to be URL encoded so I'm not sure if there's a difference in practice.  I'm guessing our default is unencoded but @pakrym could correct me if I'm wrong.  HTTP header names are case insensitive.  Query parameter names are case sensitive, but I'm guessing we erred on the side of overabundant caution so nothing gets leaked if customers accidentally specified the wrong case (and again, I don't know of any distinct Azure query parameter names that are the same modulo case so I don't think it's a problem in practice).

Reply:
I never really thought about urlencoding the query parameter names so both the UrlBuilder and sanitizer expect already pre-encoded names. 

>Query parameter names are case sensitive, but I'm guessing we erred on the side of overabundant caution so nothing gets leaked if customers accidentally specified the wrong case (and again, I don't know of any distinct Azure query parameter names that are the same modulo case so I don't think it's a problem in practice).

Yep, we are yet to see a scenario where you would like to make an opposite decision for headers that differ only in the casing. Missing a header during the sanitization feels like a more common case.

Reply:
> HTTP header names are case insensitive. 

OK, that makes sense for headers.

> but I'm guessing we erred on the side of overabundant caution so nothing gets leaked if customers accidentally specified the wrong case (and again, I don't know of any distinct Azure query parameter names that are the same modulo case so I don't think it's a problem in practice).

Regarding query parameters, I would have expected the opposite, if we wanted to avoid accidental leaks. Wouldn't having strict matching in the allow list ensure that some other parameters don't get leaked that weren't explicitly allowed? With case insensitive matching, we are allowing more today.

Reply:
You're right, sorry, these are allow lists rather than deny lists.  I'm guessing we just did that for usability then.  I'm not particularly worried about it though because I've never seen distinct query parameter names in Azure that only differ by case.  The REST API review board would never allow them through.